contracts/TiramisuSavingsClub.sol

```diff
@@ -144,7 +148,67 @@ contract TiramisuSavingsClub {
     /// @return group id
     function getGroupId() private view returns (uint) {
         uint _groupId = memberToGroupId[msg.sender];
-        require(_groupId > 0, "Caller not member of a group");
+        require(_groupId > 0, "Caller is not a member of a group");
         return _groupId;
     }
+    
+    /// @notice Checks if caller is a member of a group    
+    /// @return boolean value indicating if caller belongs to group
+    function isMemberOfGroup() public view returns (bool){            
```

Comment:
I think these ```public view``` functions are useful helpers with utility, but I would think we should strive to push complexity out of the smart contract and into the frontend wherever possible.

I wonder if a few of these helpers could be removed from the smart contract and the front end could just derive the same information for gas optimization purposes.

But I also am not sure if we should be trying to gas optimize at this stage (they say premature optimization is the root of all evil, so feel free to ignore me here :))

Reply:
I still think that we should generally strive to push complexity out of the smart contract whenever possible, but I don't think we need to do this right away.

Marking as closed